src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/InProcess/EditorInProcess_Outlinning.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Outlining;
+
+namespace Microsoft.VisualStudio.Extensibility.Testing
+{
+    internal partial class EditorInProcess
+    {
+        public async Task<ICollapsible[]> GetOutlineRegionsAsync(Text.Editor.IWpfTextView textView, CancellationToken cancellationToken)
+        {
+            var span = new SnapshotSpan(textView.TextSnapshot, 0, textView.TextSnapshot.Length);
+            var manager = await GetOutlineManagerAsync(textView, cancellationToken);
+
+            var outlines = manager.GetAllRegions(span);
+
+            return outlines
+                .OrderBy(s => s.Extent.GetStartPoint(textView.TextSnapshot))
+                .ToArray();
+        }
+
+        public async Task WaitForOutlineRegionsAsync(CancellationToken cancellationToken)
```

Comment:
I wonder if this is right. Anecdotally I've seen regions use the fallback logic prior to other computations being available. That means some regions might be available, even if they aren't the right ones :(